plugin/backend.go

```diff
@@ -106,6 +106,8 @@ func (pm *Manager) Push(name string, metaHeader http.Header, authConfig *types.A
 	if err != nil {
 		return err
 	}
+	defer config.Close()
```

Comment:
This is closed inside `Push` but it's done imperfectly. There are cases where `Push` returns without closing `rootfs` and `config`. We may want to change `Push` to take `io.Reader`s and do the closing of `config` and `rootfs` in this function.